### PR TITLE
Report invalid constructor property promotion for readonly props already defined in parent class constructor

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/8101

### CODE REVIEW

File: src/Rules/Properties/OverridingPropertyRule.php [hunk 1]
```diff
@@ -76,6 +76,14 @@ public function processNode(Node $node, Scope $scope): array
 					$prototype->getDeclaringClass()->getDisplayName(),
 					$node->getName(),
 				))->nonIgnorable()->build();
+			} else {
+				$errors[] = RuleErrorBuilder::message(sprintf(
```
Comment (staabm) [inline src/Rules/Properties/OverridingPropertyRule.php:80]: I was not able to come up with a similar example for the `static` case above this IF-block, therefore only added here.

**Comment by staabm** [comment]
@herndlm do we need a similar fix for phpdoc based readonly properties?

**Comment by herndlm** [comment]
> @herndlm do we need a similar fix for phpdoc based readonly properties?

hmm, the phpdoc-based-readonly rules are basically duplicates of the native readonly ones. but there is no equivalent for `OverridingPropertyRule` and it does not deal with phpdoc-based-readonly yet. The question is then if this is needed. And if yes if it should be added here directly (most likely just (via `isReadOnlyByPhpDoc()`) or in a dedicated rule 🤔 but either way, nothing you have to do here IMO :)

File: src/Rules/Properties/OverridingPropertyRule.php (hunk 1 above)
Comment (staabm) [reply src/Rules/Properties/OverridingPropertyRule.php:80]: just read about https://github.com/phpstan/phpstan-src/commit/3e383fc9620a46173651402049a8e8c4904591b6 which resolves my questions, as readonly properties cannot be static.

**Comment by ondrejmirtes** [comment]
I can't merge this, because it's allowed by PHP: https://3v4l.org/jPiT7

See: https://github.com/phpstan/phpstan/issues/8101#issuecomment-1268552903

We have to come up with a more clever fix for this. To realize that the property is already assigned (it's promoted) when calling the constructor.

**Pull request closed without merging**